### PR TITLE
Issue 114 - Use Camel Case in identifiers

### PR DESCRIPTION
**Pull request by chutch**

### CODE REVIEW

File: src/test/java/com/cognifide/aemrules/htl/checks/CapitalizationCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.htl.checks;
+
+import com.cognifide.aemrules.htl.AbstractBaseTest;
+import org.junit.Test;
+
+public class CapitalizationCheckTest extends AbstractBaseTest {
```
Comment (chutch) [inline src/test/java/com/cognifide/aemrules/htl/checks/CapitalizationCheckTest.java]: Please keep naming convention.
```suggestion
public class CamelCaseCheckTest extends AbstractBaseTest {
```

File: README.md [hunk 2]
```diff
@@ -126,6 +126,12 @@ Below you will find descriptions of all rules available in **AEM Rules for Sonar
 - **AEM-16** Optional is defined as ``DefaultInjectionStrategy``
   - Usage of ``@Optional`` annotation is redundant, when ``defaultInjectionStrategy`` is ``OPTIONAL``.
 
+## [HTL](https://github.com/Adobe-Marketing-Cloud/htl-spec/blob/master/SPECIFICATION.md) related
```
Comment (chutch) [inline README.md]: I'd put this section as the first one to promote new functionality.

File: src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.StringUtils;
+import org.apache.sling.scripting.sightly.compiler.expression.Expression;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.node.Attribute;
+import org.sonar.plugins.html.node.TagNode;
+
+@AemVersion(
```
Comment (chutch) [inline src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java]: Please keep convention of the order.

File: src/test/java/com/cognifide/aemrules/htl/checks/CapitalizationCheckTest.java (hunk 1 above)
Comment (piotr-wilczynski) [reply src/test/java/com/cognifide/aemrules/htl/checks/CapitalizationCheckTest.java]: fixed

File: src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java (hunk 3 above)
Comment (piotr-wilczynski) [reply src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java]: fixed

File: README.md (hunk 2 above)
Comment (piotr-wilczynski) [reply README.md]: moved

File: src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            String attributeName = attribute.getName();
+            String variableName = StringUtils.substringAfter(attributeName, ".");
+            if (StringUtils.isNotEmpty(variableName)) {
+                checkVariableName(variableName, attribute);
+            }
+        }
+    }
+
+    private void checkVariableName(@Nonnull String name, Attribute node) {
+        boolean isCamelCase = name.matches(CAMEL_CASE_PATTERN);
```
Comment (michaltobiasz) [inline src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java]: I would avoid using this method which compiles every time same pattern. Create your own Pattern as static one

File: src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+@Metadata(
+    technicalDebt = "5min"
+)
+public class CamelCaseCheck extends AbstractHtlCheck {
+
+    public static final String RULE_KEY = "HTL-3";
+
+    static final String RULE_MESSAGE = "Use Camel Case in identifiers";
+
+    private static final String CAMEL_CASE_PATTERN = "([a-z]+[A-Z]+\\w+)+";
```
Comment (michaltobiasz) [inline src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java]: If you don't need to capture the group () then use ?: ( it means that the group is matched but is not captured for back-referencing)  (?:[a-z]+[A-Z]+\\w+)+

File: src/test/files/checks/htl/CamelCaseCheck.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+<sly data-sly-use.galleryModel="com.example.GalleryModel"></sly>
+<sly data-sly-use.model="com.example.GalleryModel"></sly>
+<sly data-sly-use.GalleryModel="com.example.GalleryModel"></sly> <!--/* Non-Compliant */-->
+<sly data-sly-use.gallery_model="com.example.GalleryModel"></sly> <!--/* Non-Compliant */-->
+<sly data-sly-use.gallery-model="com.example.GalleryModel"></sly> <!--/* Non-Compliant */-->
+<sly data-sly-use.gallery_Model="com.example.GalleryModel"></sly> <!--/* Non-Compliant */-->
+<sly data-sly-use.gallery-Model="com.example.GalleryModel"></sly> <!--/* Non-Compliant */-->
+<sly data-sly-use.DAMAsset="com.example.DamAsset"></sly> <!--/* Non-Compliant */-->
+<sly data-sly-use.damAsset="com.example.DamAsset"></sly>
+<sly data-sly-use.assetFromDAM="com.example.DamAsset"></sly> <!--/* Non-Compliant */-->
```
Comment (michaltobiasz) [inline src/test/files/checks/htl/CamelCaseCheck.html]: Is this test works with current pattern?

File: src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+)
+@AemVersion(
+    from = "6.0"
+)
+@Metadata(
+    technicalDebt = "5min"
+)
+public class CamelCaseCheck extends AbstractHtlCheck {
+
+    public static final String RULE_KEY = "HTL-3";
```
Comment (michaltobiasz) [inline src/main/java/com/cognifide/aemrules/htl/checks/CamelCaseCheck.java]: The Rule key is not same like in description:
https://github.com/Cognifide/AEM-Rules-for-SonarQube/pull/137/files#diff-04c6e90faac2675aa89e2176d2eec7d8

File: src/test/files/checks/htl/CamelCaseCheck.html (hunk 6 above)
Comment (jplucinski) [reply src/test/files/checks/htl/CamelCaseCheck.html]: I've fixed regexp.

**Comment by jplucinski** [comment]
Defined **htl-15**

**Pull request merged**